File: 23-09/4.py
def firstStable(a, n, tolerance):
    for i in range(n - 1):
        diff = abs(a[i + 1] - a[i])
        
        # Add epsilon (1e-9) to avoid floating point issues
        if diff <= tolerance + 1e-9:
            return i
            
    return -1

n = int(input("Enter n: "))
    
    # Read n float values
readings = list(map(float, input("Enter the readings: ").split()))
    
tolerance = float(input("Enter tolerance: "))
    
    # Print result index
print(firstStable(readings, n, tolerance))

File: 23-09/4.c
#include<stdio.h>
#include<string.h>
int firstStable(double a[], int n, double tolerance){
	if(a.len()!=n)
{
	printf("Enter exact number of entries");
	}
{
	for(int i=0;i<n;i++)
{
	if(a[i+1] - a[i]| <= tolerance){
		return i;}
}       else
{
	return -1;
}
}
int main()
{
	int n;
	double a[],tolerance;
	printf("Enter n: ");
	scanf("%d",$n);
	printf("Enter the readings: ");
	for(int i=0;i<n;i++)
	{
	scanf("%lf",&a[i]);
	}
	printf("Enter tolerance: ")
	scanf("%lf",&tolerance);
	printf("%d",firstStable(a,n,tolerance));
	return 0;
	}

